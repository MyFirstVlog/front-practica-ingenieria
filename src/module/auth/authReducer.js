import { types } from "../../constants/types";

export const authReducer = (state, action) => {
	switch (action.type) {
		case types.login:
			return {
				...state,
				logged: true,
				email: action.payload.email,
			};

		case types.logout:
			return {
				...state,
				logged: false,
				email: null,
			};

		default:
			return state;
	}
};
