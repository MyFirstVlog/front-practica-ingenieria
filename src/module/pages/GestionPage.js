import React from "react";

export const GestionPage = () => {
  return <div>GestionPage</div>;
};
