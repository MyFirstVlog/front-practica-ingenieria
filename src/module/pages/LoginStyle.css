.login-container {
  display: flex;
  justify-content: center;
  align-items: center;
}

.login-body {
  position: relative;
  height: 100vh;
  width: 414px;
}

.login-image {
  position: absolute;
  display: flex;
  justify-content: center;
  top: 0;
  height: 70vh;
  width: 100%;
  background-image: url("https://res.cloudinary.com/dpp07uuh5/image/upload/v1655347834/Grupo_41_2x_cujp7d.png");
  background-size: cover;
}

.logo {
  margin-top: 20px;
  width: 100px;
  height: 60px;
  justify-content: center;
}

.bottom-card {
  position: absolute;
  padding: 30px;
  display: flex;
  flex-direction: column;
  bottom: 0;
  height: 40vh;
  width: 100%;
  background-color: #2D3A75;
  border-radius: 40px 40px 0 0;
}

.container {
  display: flex;
  justify-content: center;
  margin-bottom: auto;
}

.access-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  width: fit-content;
}

.name-container {
  display: flex;
  flex-direction: column;
  width: 70%;
}

.name-container span{
  color: white;
  font-size: 16px;
  text-align: center;
  margin-top: -4px;
}

.email-login-container{
  color: white;
  font-size: 14px;
  margin-top: -10px;
  margin-left: 10px;
}

.google-title{
  margin-left: 10px;
}

@media (max-width: 414px) {
  .login-body {
    position: relative;
    height: 100vh;
    width: 100%;
  }
}

@media (max-width: 380px) {
  .bottom-card {
    padding: 10px;
    padding-top: 30px;
  }

  .name-container {
    width: 90%;
  }

}