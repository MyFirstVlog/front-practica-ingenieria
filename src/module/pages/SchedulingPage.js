import React from "react";

export const SchedulingPage = () => {
  return <div>Page</div>;
};
