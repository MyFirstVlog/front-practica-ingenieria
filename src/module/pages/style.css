:root {
    --primary: #2d3a75;
    --secondary: #426aa7;
    --accent: #4db1e0;
    --gray: #ededed;
    --gray-primary: #434345;
    --gray-medium: #6f6e75;
    --gray-secondary: #cccccc;
}

.form-container {
    width: 30%;
    margin: auto;
    padding-bottom: 24px !important;
    display: block;
    background: var(--gray);
}

.card {
    border-radius: 15px;
    box-shadow: 0 1px 3px rgb(0 0 0 / 12%), 0 1px 2px rgb(0 0 0 / 24%);
    transition: all .3s cubic-bezier(.25, .8, .25, 1);
}

.form-header {
    padding: 8px 0;
}

.img-cidenet {
  max-width: 30%;
  height: auto;
  padding: 20px 20px 20px 20px;
}

.bcg-login {
    background-color: var(--gray-secondary);
}

.img-test-cidenet {
    width: 35%;
    margin: auto;
    display: block;
    background-image: url("https://res.cloudinary.com/dpp07uuh5/image/upload/v1655347834/Grupo_41_2x_cujp7d.png");
    /* background-position: center; */
    background-repeat: no-repeat;
    background-size: cover;
    height: 100vh;
    border-radius: 35px 35px 35px 35px;
}

.center-items {
    display: flex;
    justify-content: center;
}

.primer-div {
    height: 60%;
}

.segundo-div {
    height: 40%;
    background-color: var(--primary);
    border-radius: 35px 35px 0px 0px;
}

.title-gestionappte {
    padding: 45px 45px 45px 45px;
    color: var(--gray);
    display: flex;
    justify-content: center;
}

.title-google-login {
    padding: 40px 60px 40px 150px;
}

.font-size-large {
    font-size: large;
}

.font-size-medium {
    font-size: medium;
}

.white-color {
    color: var(--gray);
}
.margin-bottom {
    margin-bottom: 15px;
}
.btn-login {
    color: var(--primary);
    background-color: var(--gray);
}
.btn-login-content {
    display: flex;
    justify-content: space-between;
}


/* .auth__social-networks{
    align-items: center;
    display: flex;
    justify-content: center;
    flex-direction: column;
    padding-top: 20px;
    padding-bottom: 20px;
    width: 100%;
}

.google-btn {
    cursor: pointer;
    margin-top: 5px;
    width: 100%;
    height: 42px;
    background-color: #4285f4;
    border-radius: 2px;
    box-shadow: 0 3px 4px 0 rgba(0, 0, 0, 0.25);

    transition: box-shadow .3s ease;
}
.google-icon-wrapper {
    position: absolute;
    margin-top: 1px;
    margin-left: 1px;
    width: 40px;
    height: 40px;
    border-radius: 2px;
    background-color:#fff;
}
.google-icon {
    position: absolute;
    margin-top: 11px;
    margin-left: 11px;
    width: 18px;
    height: 18px;
}

.btn-text {
    float: right;
    margin: 11px 40px 0 0;
    color:rgb(0, 0, 0);
    font-size: 14px;
    letter-spacing: 0.2px;
} */
