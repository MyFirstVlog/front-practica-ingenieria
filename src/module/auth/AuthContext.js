import React from "react";
import { createContext, useReducer } from "react";
import { authReducer } from "./authReducer";

export const AuthContext = createContext();

const initialState = {
	logged: false,
	email: "",
};

export const AuthProvider = ({ children }) => {
	const [authState, dispatch] = useReducer(authReducer, initialState);

	return (
		<AuthContext.Provider
			value={{
				authState,
				dispatch,
			}}
		>
			{children}
		</AuthContext.Provider>
	);
};
