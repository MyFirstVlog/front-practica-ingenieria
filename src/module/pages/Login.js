import React from 'react';
import './LoginStyle.css';
import './style.css';

export const LoginPage = () => {

  return (
    <div className="login-container">
      <div className="login-body">
        <div className="login-image">
          <img
            src="https://res.cloudinary.com/dpp07uuh5/image/upload/v1655347814/atomo-logo_2x_k5gsej.png"
            alt="cidenet"
            className="logo"
          ></img>
        </div>
        <div className="bottom-card">
          <div className="container">
            <div className="name-container">
              <img
                src="https://res.cloudinary.com/dpp07uuh5/image/upload/v1655439104/atomo-logogestionappte_2x_y4ljlf.png"
                alt="cidenet"
              ></img>
              <span>Para nuestros colaboradores</span>
            </div>
          </div>
          <div className="container">
            <div className="access-container">
              <div className="col font-size-medium margin-bottom google-title">
                <strong className="white-color">Iniciar sesión</strong>
              </div>
              <div className="row">
                <div className="col margin-bottom">
                  <button className="btn btn-login rounded-pill">
                    <div className="row">
                      <div className="col ">
                        <i className="fa-brands fa-google"></i> |
                        <strong> INICIA SESIÓN CON GOOGLE</strong>
                      </div>
                    </div>
                  </button>
                </div>
              </div>
              <div className="email-login-container">
                <span>Ingresa con el correo corporativo</span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};