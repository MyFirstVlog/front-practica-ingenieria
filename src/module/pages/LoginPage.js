import React from 'react';
// import { types } from '../../constants/types';
// import { AuthContext } from '../auth/AuthContext';
// import { useForm } from '../hooks/useForm';
import './style.css';

export const LoginPage = () => {
	/*const [available, setAvailable] = useState(true);
	const [values, handleInputChange] = useForm({
		email: '',
		password: '',
	});
	const { dispatch } = useContext(AuthContext);

	const onSubmit = (e) => {
		e.preventDefault();
		if (values.email && values.password) {
			dispatch({
				type: types.login,
				payload: {
					email: values.email,
				},
			});
		}
	};

	useEffect(() => {
		if (values.email && values.password) {
			setAvailable(false);
		}
	}, [values]);*/

	return (
		<div className="bcg-login">
			<div className="img-test-cidenet">
				<div className="primer-div">
					<div className="row">
						<div className="col center-items">
							<img
								src="https://res.cloudinary.com/dpp07uuh5/image/upload/v1655347814/atomo-logo_2x_k5gsej.png"
								alt="cidenet"
								className="img-cidenet"
							></img>
						</div>
					</div>
				</div>
				<div className="segundo-div">
					{/* <div className="row title-gestionappte">
						<div className="col center-items">Para nuestros colaboradores</div>
					</div> */}
					<div className="row title-gestionappte">
						<div className="row">
							<div className="col center-items">
								<img
									src="https://res.cloudinary.com/dpp07uuh5/image/upload/v1655439104/atomo-logogestionappte_2x_y4ljlf.png"
									alt="cidenet"
									className="gestionappte-cidenet"
									style={{ maxWidth: '55%' }}
								></img>
							</div>
						</div>
						<div className="row">
							<div className="col center-items font-size-large">
								<span>Para nuestros colaboradores</span>
							</div>
						</div>
						{/* <div className="col">hola mundo</div> */}
					</div>
					<div className=" row title-google-login">
						<div className="row">
							<div className="col font-size-large margin-bottom">
								<strong className="white-color">Iniciar sesión</strong>
							</div>
						</div>
						<div className="row">
							<div className="col margin-bottom">
								<button className="btn btn-login rounded-pill">
									<div className="row">
										<div className="col ">
											<i className="fa-brands fa-google"></i> |
											<strong> INICIA SESIÓN CON GOOGLE</strong>
										</div>
									</div>
								</button>
							</div>
						</div>
						<div className="row">
							<div className="col margin-bottom white-color">
								<span>Ingresa con el correo corporativo</span>
							</div>
						</div>
					</div>
				</div>
			</div>
		</div>
	);
};

// <div className="mt-5 card form-container">
// 	<div className="form-header text-center">
// 		<img
// 			src="https://res.cloudinary.com/crunchbase-production/image/upload/c_lpad,h_256,w_256,f_auto,q_auto:eco,dpr_1/ltgilmylvmrd2mnrsjem"
// 			alt="cidenet"
// 			className="img-cidenet"
// 		/>
// 	</div>
// 	<div className="card-body">
// 		<form onSubmit={onSubmit}>
// 			<div className="mb-4">
// 				<label htmlFor="exampleInputEmail1" className="form-label">
// 					Email
// 				</label>
// 				<input
// 					type="email"
// 					className="form-control"
// 					id="exampleInputEmail1"
// 					aria-describedby="emailHelp"
// 					name="email"
// 					onChange={handleInputChange}
// 				/>
// 			</div>
// 			<div className="mb-4">
// 				<label htmlFor="exampleInputPassword1" className="form-label">
// 					Contraseña
// 				</label>
// 				<input
// 					type="password"
// 					className="form-control"
// 					id="exampleInputPassword1"
// 					name="password"
// 					onChange={handleInputChange}
// 				/>
// 			</div>
// 			<div className="text-center">
// 				<button
// 					type="submit"
// 					className="btn btn-primary mb-2 rounded-pill"
// 					disabled={available}
// 				>
// 					<span style={{ marginRight: '7px' }}>
// 						<i className="fa-brands fa-google"></i>
// 					</span>
// 					INICIA SESIÓN CON GOOGLE
// 				</button>
// 			</div>
// 		</form>

{
	/* <div className="auth__social-networks">
					<div className="google-btn">
						<div className="google-icon-wrapper">
							<img
								className="google-icon"
								src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
								alt="google button"
							/>
						</div>
						<p className="btn-text">
							<b>Sign in with google</b>
						</p>
					</div>
				</div> */
}

{
	/* <div className='auth-testing'>

				</div> */
}
{
	/* <h5 className="card-title">Special title treatment</h5>
				<p className="card-text">
					With supporting text below as a natural lead-in to additional content.
				</p>
				<button onClick={onClick} className="btn btn-primary">
					Go somewhere
				</button> */
}

{
	/* </div>
		</div> */
}
