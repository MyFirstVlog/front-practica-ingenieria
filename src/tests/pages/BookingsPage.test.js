import React from "react";
import { mount } from "enzyme";
import { MemoryRouter, Router } from "react-router-dom";
import "@testing-library/jest-dom";
import { AuthContext } from "../../module/auth/AuthContext";
import { BookingsPage } from "../../module/pages/BookingsPage";

describe("should test NavBar component", () => {
	const historyMock = {
		push: jest.fn(),
		replace: jest.fn(),
		location: {},
		listen: jest.fn(),
		createHref: jest.fn(),
	};

	const contextValue = {
		dispatch: jest.fn(),
		authState: {
			logged: true,
			email: "example1@example.com",
		},
	};
	const wrapper = mount(
		<AuthContext.Provider value={contextValue}>
			<MemoryRouter>
				<Router history={historyMock}>
					<BookingsPage />
				</Router>
			</MemoryRouter>
		</AuthContext.Provider>
	);

	afterEach(() => {
		jest.clearAllMocks();
	});
	test("should test BookingsPage", () => {
		expect(wrapper).toBeDefined();
		expect(wrapper).toMatchSnapshot();
	});
});
