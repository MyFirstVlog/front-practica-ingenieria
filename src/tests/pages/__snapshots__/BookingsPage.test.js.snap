// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test NavBar component should test BookingsPage 1`] = `
<div>
  BookingsPage
</div>
`;
